### DISCUSSION TITLE
Multi Cursor copy & paste

### DISCUSSION THREAD

**Discussion by noor-tg** [helix-editor/helix#5674] [Q&A]
hello,
I want to select multiple text in document. copy the selected text. and paste it to another location.

when i tried to select & copy & paste . only the first text selected in the multi select is pasted.

how to do this in helix ?

**Comment by Proful** [comment]
The below behaviour u r looking for?

- Shift+C for multiple cursor
- e to select word
- y to yank
- move both cursor to down
- p to paste
- esc

![multi_select_copy_paste](https://user-images.githubusercontent.com/354596/214521692-9a762955-da87-4a94-8270-dfeadbda1e98.gif)

**Reply by noor-tg** [reply]
what if I want to copy the content to other frame? so the multi selection is not there when switching to the other frame.

**Reply by hellocode808** [reply]
I am trying to wrap my head around the multicursor way of doing things and thought the following might be a good use case for it. However, I can't figure out how.

I have to following SQL Tables and want to select and paste the column tables from the first table to a different buffer. 

```sql
CREATE TABLE IF NOT EXISTS "sessions" (
  id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,
  client_id INTEGER NOT NULL,
  date DATE,
  text TEXT,
  has_invoice INTEGER, 
  FOREIGN KEY (client_id)
    REFERENCES clients (id)
);

CREATE TABLE IF NOT EXISTS "services" (
  id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,
  name TEXT,
  description TEXT,
  created_at DATETIME,
  deleted_at DATETIME,
  default_price INTEGER,
  custom INTEGER
);
```

I got this far:

- set cursor on 'i' from id
- select with "4 Shift-C e"
- yank with "Space-y"

However, the pasting either returns just the value of the first cursor or when I do it with "Space-p" it always messes up my other code below with the yanked values.  

Can I join the yanked values somehow to one yank that I can then paste in my other buffer?

**Reply by gabydd** [reply]
You can use the ``:yank-join`` typeable command

**Reply by hellocode808** [reply]
Perfect! Thank you.

**Reply by thiago-figueredo** [reply]
@Proful which font are you using in the image ?

**Comment by CptPotato** [comment]
Alternatively, you can (ab)use the system clipboard:

- join and yank multiple selections: <kbd>space</kbd><kbd>y</kbd>
- paste: <kbd>space</kbd><kbd>p</kbd> (or <kbd>space</kbd><kbd>P</kbd>)

**Edit:** This doesn't work out of the box anymore, unless <kbd>space</kbd><kbd>y</kbd> is remapped to `:clipboard-yank-join`

**Comment by dev-hack95** [comment]
How to do copy from helix and paste it to browser

**Reply by masriomarm** [reply]
Select text then `space y`
